ttnn/cpp/ttnn/operations/data_movement/common/common.hpp

```diff
@@ -156,6 +156,65 @@ ttnn::Tensor pad_to_tile_vol(
     const bool use_multicore,
     const std::optional<MemoryConfig>& memory_config);
 
+enum class ShardStrategy { BLOCK, HEIGHT, WIDTH };
+
+ttnn::MemoryConfig create_sharded_memory_config(
+    const ttnn::Shape& shape,
+    const tt::tt_metal::CoreRangeSet& core_grid,
+    const ShardStrategy& strategy,
+    const tt::tt_metal::ShardOrientation& orientation,
+    bool halo = false,
+    bool use_height_and_width_as_shard_shape = false,
+    const tt::tt_metal::Layout& layout = tt::tt_metal::Layout::ROW_MAJOR);
+
+template <ShardStrategy strategy>
+std::pair<uint32_t, std::vector<uint32_t>> tensor_coord_to_shard_coord(
```

Comment:
Please, let's remove a template here, and move it to cpp.

Reply:
Please change the return type.  `std::vector<uint32_t>` should be an `std::array<uint32_t, 2>`, and the same thing for `shard_shape` argument

Reply:
Done.